docs/workflow/development-workflow.md

```diff
@@ -0,0 +1,269 @@
+
+If the IDE was configured well, it is even possible for it to provide prompts for function parameters and reference to the appropriate header files when looking up the definition of a particular function call.
+This is another useful feature especially when dealing with custom libraries and APIs (not the standard language supplied library APIs)
+
+## Adoption of Common Style Guidelines
+
+Very often we start from example projects or code taken from other open source projects. This is the basis for ev3dev after all. While it is tempting to just add new code to the project willy-nilly, it is better to try and be consistent with the existing project style guidelines, to ensure readability as well as future sharing of the code with others. 
+
+> Large projects frequently have strict style guidelines and will not accept code contribution from others which do not meet those guidelines.
+
+## Using Source Revision Management Software
+
+Source code control, revision management, or code control, refer to the task of managing changes to the program code over time. The ability to roll back a change which introduced an unexpected bug, or to test out new features while the already working code is being used actively, is a powerful feature of Source Revision Management Software. Beyond these typical features that benefit an individual programmer tremendously, almost all source revision management software also allows collaboration with other programmers to improve the code and work on larger projects in a team.
+
+Since this is a huge topic in itself, this guide will not get into further details, except to note that knowledge of commonly-used source revision management software is a useful skill for any programmer to know, and is well worth the time investment involved. If you're new to this whole concept of source code control, [Git](https://git-scm.com/) is the go-to software for source revision management today.
+
+>For reference, all ev3dev projects are using Git for source code control; Git is the basis for all projects hosted on GitHub as well.
+
+# Downloading Programs to Target
+
+* Enabling Networking on the Target
+* Setting Up SSH Access
+* Using Public Key Authentication
+* Downloading/copying Programs to the Robot Controller
+
+## Enabling Networking on the Target
+
+See the topic on [Networking](http://www.ev3dev.org/docs/networking/) for more information on how to enable network access on the Robot Controller.
+
+## Setting up SSH Access
```

Comment:
This section seems to be duplicated.

Reply:
The Networking part is necessary since I'd rather not describe it in detail here.

The existing description of setting up SSH is rather brief. I guess what can be done is to move this part to the SSH page. What I wanted to highlight was the Public Key Authentication part which is useful to know.